weatherman.py

```diff
@@ -0,0 +1,201 @@
+        'most_humidity_day': None,
+        'most_humidity': 0
+    }
+
+    for weather_record in weather_records:
+        maximum_temperature = weather_record['Max TemperatureC']
+        minimum_temperature = weather_record['Min TemperatureC']
+        humidity = weather_record['Max Humidity']
+
+        try:
+            if maximum_temperature and float(maximum_temperature) > yearly_weather_record['highest_temperature']:
+                yearly_weather_record['highest_temperature'] = float(maximum_temperature)
+                yearly_weather_record['highest_temperature_day'] = weather_record['PKT']
+
+            if minimum_temperature is not None and float(minimum_temperature) < yearly_weather_record['lowest_temperature']:
+                yearly_weather_record['lowest_temperature'] = float(minimum_temperature)
+                yearly_weather_record['lowest_temperature_day'] = weather_record['PKT']
+
+            if humidity and float(humidity) > yearly_weather_record['most_humidity']:
+                yearly_weather_record['most_humidity'] = float(humidity)
+                yearly_weather_record['most_humidity_day'] = weather_record['PKT']
+        except ValueError:
+            continue
+
+    return yearly_weather_record
+
+
+def calculate_monthly_weather_record(weather_records):
+    month_weather_record = {
+        'mean_humidity': 0,
+        'mean_iteration': 0,
+        'average_lowest_temperature': 0,
+        'average_highest_temperature': 0
+    }
+
+    for weather_record in weather_records:
+        mean_humidity_weather_record = weather_record[' Mean Humidity']
+        maximum_temperature = weather_record['Max TemperatureC']
+        minimum_temperature = weather_record['Min TemperatureC']
+
+        try:
+            if mean_humidity_weather_record and float(mean_humidity_weather_record) > 0:
+                month_weather_record['mean_humidity'] += float(mean_humidity_weather_record)
+                month_weather_record['mean_iteration'] += 1
+
+            if maximum_temperature and float(maximum_temperature) > 0:
+                month_weather_record['average_highest_temperature'] += float(maximum_temperature)
+
+            if minimum_temperature and float(minimum_temperature) > 0:
+                month_weather_record['average_lowest_temperature'] += float(minimum_temperature)
+        except ValueError:
+            continue
+
+    return month_weather_record
+
+
+def print_yearly_weather_record(highest_temperature, highest_temperature_day, lowest_temperature, lowest_temperature_day, most_humidity, most_humidity_day):
+    print(f"Highest Temperature: {highest_temperature}°C on {highest_temperature_day}")
+    print(f"Lowest Temperature: {lowest_temperature}°C on {lowest_temperature_day}")
+    print(f"Most humidity {most_humidity}% on  {most_humidity_day} ")
+
+
+def print_monthly_weather_record(mean_iteration, mean_humidity, average_lowest_temperature, average_highest_temperature):
```

Comment:
Shouldn't we use a more suitable data structure rather than passing a large number of individual values?